src/Idris/Elab/Class.hs

```diff
@@ -164,12 +164,12 @@ elabClass info syn_in doc fc constraints tn tnfc ps pDocs fds ds mcn cd
     checkDefaultSuperclassInstance :: PDecl -> Idris ()
     checkDefaultSuperclassInstance (PInstance _ _ _ fc cs _ _ _ n _ ps _ _ _ _)
         = do when (not $ null cs) . tclift
-                $ tfail (At fc (Msg "Default superclass instances can't have constraints."))
+                $ tfail (At fc (Msg "Default superinterface instances can't have constraints."))
```

Comment:
I am not sure "superinterface" is a valid English word. Perhaps use "super-interface" or "super interface" instead?